libcloud/dns/drivers/godaddy.py

```diff
@@ -0,0 +1,498 @@
+                     name, you should specify empty string ('') for this
+                     argument.
+        :type  name: ``str``
+
+        :param zone: Zone where the requested record is created.
+        :type  zone: :class:`Zone`
+
+        :param type: DNS record type (A, AAAA, ...).
+        :type  type: :class:`RecordType`
+
+        :param data: Data for the record (depends on the record type).
+        :type  data: ``str``
+
+        :param extra: Extra attributes (driver specific). (optional)
+        :type extra: ``dict``
+
+        :rtype: :class:`Record`
+        """
+        new_record = self._format_record(name, type, data, extra)
+        self.connection.request(
+            '/v1/domains/%s/records' % (zone.domain), method='PATCH',
+            data=[new_record])
+        id = self._get_id_of_record(name, type)
+        return Record(
+            id=id, name=name,
+            type=type, data=data,
+            zone=zone, driver=self,
+            ttl=new_record['ttl'],
+            extra=extra)
+
+    def update_record(self, record, name, type, data, extra=None):
+        """
+        Update an existing record.
+
+        :param record: Record to update.
+        :type  record: :class:`Record`
+
+        :param name: Record name without the domain name (e.g. www).
+                     Note: If you want to create a record for a base domain
+                     name, you should specify empty string ('') for this
+                     argument.
+        :type  name: ``str``
+
+        :param type: DNS record type (A, AAAA, ...).
+        :type  type: :class:`RecordType`
+
+        :param data: Data for the record (depends on the record type).
+        :type  data: ``str``
+
+        :param extra: (optional) Extra attributes (driver specific).
+        :type  extra: ``dict``
+
+        :rtype: :class:`Record`
+        """
+        new_record = self._format_record(name, type, data, extra)
+        self.connection.request(
+            '/v1/domains/%s/records' % (record.zone.domain), method='PUT',
+            data=[new_record])
+        id = self._get_id_of_record(name, type)
+        return Record(
+            id=id, name=name,
+            type=type, data=data,
+            zone=record.zone, driver=self,
+            ttl=new_record['ttl'],
+            extra=extra)
+
+    def get_record(self, zone_id, record_id):
+        """
+        Return a Record instance.
+
+        :param zone_id: ID of the required zone
+        :type  zone_id: ``str``
+
+        :param record_id: ID of the required record
+        :type  record_id: ``str``
+
+        :rtype: :class:`Record`
+        """
+        parts = record_id.split(':')
+        result = self.connection.request(
+            '/v1/domains/%s/records/%s/%s' % (
+                zone_id,
+                parts[1],
+                parts[0])).object
+        if len(result) == 0:
+            raise RecordDoesNotExistError()
```

Comment:
I don't think this will work - you need to pass the following arguments to the constructor - value, driver, record_id